### PR TITLE
[13.0][IMP] helpdesk_mgmt: User: Team tickets group

### PR DESCRIPTION
**Pull request by victoralmau**
User: Team tickets group

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT40878

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Can you update the readme with this new setting?

**Review by chienandalu — APPROVED** [review]

File: helpdesk_mgmt/README.rst [hunk 1]
```diff
@@ -115,6 +115,8 @@ Tags
 Usage
 =====
 
+#. Different user permissions are available to restrict access to tickets: User: Personal tickets, User: Team tickets and User.
```
Comment (pedrobaeza) [inline helpdesk_mgmt/README.rst]: This is configuration, not usage, and please explain the goal of each permission.

File: helpdesk_mgmt/models/helpdesk_ticket_team.py [hunk 2]
```diff
@@ -9,7 +9,13 @@ class HelpdeskTeam(models.Model):
     _inherit = ["mail.thread", "mail.alias.mixin"]
 
     name = fields.Char(string="Name", required=True)
-    user_ids = fields.Many2many(comodel_name="res.users", string="Members")
+    user_ids = fields.One2many(
```
Comment (pedrobaeza) [inline helpdesk_mgmt/models/helpdesk_ticket_team.py]: You can belong to more than one team, so this is not correct.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: helpdesk_mgmt/README.rst (hunk 1 above)
Comment (victoralmau) [reply helpdesk_mgmt/README.rst]: IMO it is not necessary to add this because I think the name of the groups is obvious, what do you think?

Comment (pedrobaeza) [reply helpdesk_mgmt/README.rst]: Well, it's not bad to explain a bit more apart from the name. Sometimes the obvious for us is not the same for the plain user.

File: helpdesk_mgmt/readme/CONFIGURE.rst [hunk 3]
```diff
@@ -74,3 +74,13 @@ Tags
...
    :alt: Ticket Tags
    :width: 600 px
+
+
+Permissions
+~~~~~~~~~~~
+
+There are restrictions to read tickets according to the user's permissions set in Helpdesk.
+
+#. *User: Personal tickets*: User can list his tickets (those that are assigned to his user) or those that are not assigned to anyone.
```
Comment (pedrobaeza) [inline helpdesk_mgmt/readme/CONFIGURE.rst]: Use gender neutral words:

```suggestion
#. *User: Personal tickets*: User can list their tickets (those that are assigned to their user) or those that are not assigned to anyone.
```

Comment (pedrobaeza) [reply helpdesk_mgmt/readme/CONFIGURE.rst]: And this should be that the tickets are only visible if no team nor user is assigned.

File: helpdesk_mgmt/readme/CONFIGURE.rst [hunk 4]
```diff
@@ -74,3 +74,13 @@ Tags
...
+#. *User: Team tickets*: User can list the tickets that are assigned to the teams to which he/she belongs or the tickets that are not assigned to any team.
```
Comment (pedrobaeza) [inline helpdesk_mgmt/readme/CONFIGURE.rst]: It should be that are not assigned to any team nor user.

File: helpdesk_mgmt/readme/CONFIGURE.rst [hunk 5]
```diff
@@ -74,3 +74,13 @@ Tags
...
+#. *User*: User can list all tickets.
```
Comment (pedrobaeza) [inline helpdesk_mgmt/readme/CONFIGURE.rst]: Change better to `is able to see` on all the lines.

```suggestion
#. *User*: User is able to see all the tickets.
```

File: helpdesk_mgmt/security/helpdesk_security.xml [hunk 6]
```diff
@@ -6,10 +6,15 @@
...
+        <record id="group_helpdesk_user_team" model="res.groups">
+            <field name="name">User: Team tickets</field>
+            <field name="category_id" ref="module_helpdesk_category" />
+            <field name="implied_ids" eval="[(4, ref('group_helpdesk_user_own'))]" />
+        </record>
         <record id="group_helpdesk_user" model="res.groups">
             <field name="name">User</field>
             <field name="category_id" ref="module_helpdesk_category" />
-            <field name="implied_ids" eval="[(4, ref('group_helpdesk_user_own'))]" />
+            <field name="implied_ids" eval="[(4, ref('group_helpdesk_user_team'))]" />
```
Comment (pedrobaeza) [inline helpdesk_mgmt/security/helpdesk_security.xml:17]: This change requires a migration script.

File: helpdesk_mgmt/views/helpdesk_ticket_views.xml [hunk 7]
```diff
@@ -154,7 +154,7 @@
                             <field name="user_ids" invisible="1" readonly="1" />
                             <field
                                 name="user_id"
-                                domain="[('share', '=', False)]"
+                                domain="[('share', '=', False),('id', 'in', user_ids)]"
```
Comment (pedrobaeza) [inline helpdesk_mgmt/views/helpdesk_ticket_views.xml]: Please put at field level instead, and it should be:
```
domain="team_id and [('share', '=', False),('id', 'in', user_ids)] or [('share', '=', False)]"
```

File: helpdesk_mgmt/models/res_users.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+from odoo import fields, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    helpdesk_team_ids = fields.Many2many(
```
Comment (pedrobaeza) [inline helpdesk_mgmt/models/res_users.py:8]: Why is this needed?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You should handle also if a user is assigned and you change the team, that the user should be kept only if it belongs to that team.

File: helpdesk_mgmt/models/res_users.py (hunk 8 above)
Comment (victoralmau) [reply helpdesk_mgmt/models/res_users.py:8]: This is necessary for the rule (`ir.rule`). Another possibility is to use `user_ids` field of tickets, which option do you think is better?

Comment (pedrobaeza) [reply helpdesk_mgmt/models/res_users.py:8]: We should balance performance and complexity. The question here is that this field is not computed, so how it's filled?

Comment (victoralmau) [reply helpdesk_mgmt/models/res_users.py:8]: This field comes from https://github.com/OCA/helpdesk/blob/13.0/helpdesk_mgmt/models/helpdesk_ticket_team.py#L12

Comment (pedrobaeza) [reply helpdesk_mgmt/models/res_users.py:8]: OK, please put a comment about that, and fix in both the name of the columns and table for avoiding surprises in the future.

**Comment by victoralmau** [comment]
All suggested changes done.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major
